apl/scalar-functions/array-functions/array-iff.mdx

```diff
@@ -0,0 +1,148 @@
+---
+title: array_iff
+description: 'This page explains how to use the array_iff function in APL.'
+---
+
+## Introduction
+
+The `array_iff` function in Axiom Processing Language (APL) allows you to create arrays based on a condition. It returns an array with elements from two specified arrays, choosing each element from the first array when a condition is met and from the second array otherwise. This function is useful for scenarios where you need to evaluate a series of conditions across multiple datasets, especially in log analysis, trace data, and other applications requiring conditional element selection within arrays.
+
+## For users of other query languages
+
+If you come from other query languages, this section explains how to adjust your existing queries to achieve the same results in APL.
+
+<AccordionGroup>
+<Accordion title="Splunk SPL users">
+
+In Splunk SPL, array manipulation based on conditions typically requires using conditional functions or eval expressions. APL’s `array_iff` function lets you directly select elements from one array or another based on a condition, offering more streamlined array manipulation.
+
+<CodeGroup>
+```sql Splunk example
+eval selected_array=if(condition, array1, array2)
+```
+
+```kusto APL equivalent
+array_iff(condition_array, array1, array2)
+```
+</CodeGroup>
+
+</Accordion>
+<Accordion title="ANSI SQL users">
+
+In ANSI SQL, conditionally selecting elements from arrays often requires complex `CASE` statements or functions. With APL’s `array_iff` function, you can directly compare arrays and conditionally populate them, simplifying array-based operations.
+
+<CodeGroup>
+```sql SQL example
+CASE WHEN condition THEN array1 ELSE array2 END
+```
+
+```kusto APL equivalent
+array_iff(condition_array, array1, array2)
+```
+</CodeGroup>
+
+</Accordion>
+</AccordionGroup>
+
+## Usage
+
+### Syntax
+
+```kusto
+array_iff(condition_array, array1, array2)
+```
+
+### Parameters
+
+- `condition_array`: An array of boolean values, where each element determines whether to choose the corresponding element from `array1` or `array2`.
+- `array1`: The array to select elements from when the corresponding `condition_array` element is `true`.
+- `array2`: The array to select elements from when the corresponding `condition_array` element is `false`.
+
+### Returns
+
+An array where each element is selected from `array1` if the corresponding `condition_array` element is `true`, and from `array2` otherwise.
+
+## Use case examples
+
+<Tabs>
+<Tab title="Log analysis">
+
+The `array_iff` function can help filter log data conditionally, such as choosing specific durations based on HTTP status codes.
+
+**Query**
+
+```kusto
+['sample-http-logs']
+| extend duration_conditions = array_iff(status == '200', req_duration_ms, 0)
+| project _time, duration_conditions
```

Comment:
How about something like this: https://play.axiom.co/axiom-play-qf1k/explorer?qid=pZxTWjaWAAc-sm6fpl&relative=1